modulus/metrics/climate/healpix_loss.py

```diff
@@ -189,3 +189,68 @@ def forward(self, prediction, target, average_channels=True):
             return th.sum(ocean_mean_err) / self.lsm_sum
         else:
             return ocean_mean_err / self.lsm_var_sum
+
+
+class WeightedOceanMSE( th.nn.MSELoss ):
+    """
+    Ocean MSE class offers impementaion for MSE loss with: 
+    1) weighted by a land-sea-mask field. 
+    2) weighted by channel (e.g. sic more than sst)
+    """
+    def __init__(
+        self,
+        lsm_file: str,
+        open_dict: dict = {
+            'engine':'zarr'},
+        selection_dict: dict = {
+            'channel_c':'lsm'},
+        weights: Sequence = [],
+    ):
+        """
+        """
+        super().__init__()
+        self.device = None
+        self.lsm_file = lsm_file
+        self.lsm_ds = None
+        self.open_dict = open_dict
+        self.selection_dict = selection_dict
+        self.lsm_tensor = None 
+        self.lsm_sum_calculated = False
+        self.lsm_sum = None
+        self.lsm_var_sum = None
+        self.loss_weights = th.tensor(weights)
+
+    def setup(self, trainer):
+        """
+        reshape lsm and put on device 
+        pushes weights to cuda device 
+        """
+        ### 1. OCEAN PREP ###
```

Comment:
in the recent training data we curated at Nvidia land sea mask is a channel in the zarr training data, this implies that a separate lsm file is found, is there a reason for this?

Reply:
I've implemented this to be consistent with the current implementation of OceanMSE (line 169 in this file) which also loads an external LSM file. Do you know how the DLOMs you or other NVIDIA folks have trained handled this?

I agree that eventually this should be updated for both the OceanMSE and WeightedOceanMSE implementations.

Reply:
We need to have access to the LSM within the loss class to provide weights to the loss. The way it's implemented here, we pass the path to a file which contains the LSM data.

This this file could/should be the training dataset in most cases.